jerry-core/api/jerry.c

```diff
@@ -1524,7 +1524,7 @@ jerry_create_regexp_sz (const jerry_char_t *pattern_p, /**< zero-terminated UTF-
 
   ecma_string_t *ecma_pattern = ecma_new_ecma_string_from_utf8 (pattern_p, pattern_size);
 
-  jerry_value_t ret_val = ecma_op_create_regexp_object (ecma_pattern, flags);
+  jerry_value_t ret_val = ecma_op_create_regexp_object (ecma_pattern, (uint16_t) flags);
```

Comment:
I see that there is a `re_flags_t` enum in the `ecma-regexp-object.h` should be use that type here instead (and possibly in the `ecma_op_create_regexp_object` and other relevant methods)?

Reply:
The problem is we use the combination of flags, rather than a single flag. enum -> int conversions has strange rules, some compilers use the smallest type, some compilers use int all the time.

Reply:
then the usage of the `jerry_regexp_flags_t` enum type in the function is also not correct, or not?

Reply:
good point